legate/util/system.py

```diff
@@ -98,9 +105,13 @@ def cpus(self) -> tuple[CPUInfo, ...]:
                 line = open(
                     f"/sys/devices/system/cpu/cpu{i}/topology/thread_siblings_list"  # noqa E501
                 ).read()
-                sibling_sets.add(
-                    tuple(sorted(int(x) for x in line.strip().split(",")))
-                )
+
+                flattened = []
+                for x in (
+                    expand_range(value) for value in line.strip().split(",")
+                ):
+                    flattened += [y for y in x]
+                sibling_sets.add(tuple(sorted(flattened)))
```

Comment:
`itertools.chain` could simplify here:
```python
In [10]: line = "1-3,4,5,8-12,15,20"

In [11]: tuple(sorted(chain.from_iterable(expand_range(r) for r in line.strip().split(","))))
Out[11]: (1, 2, 3, 4, 5, 8, 9, 10, 11, 12, 15, 20)
```
In fact with `chain`  you could have `expand_range` just return the `range` directly, and not even call `tuple` to materialize it

Reply:
That's amazing, done!

I'll keep the tuple to so that the type checking for pre-commit stays happy.